initramfs-init.in

```diff
@@ -435,9 +435,22 @@ fi
 
 # locate boot media and mount it
 ebegin "Mounting boot media"
-nlplug-findfs $cryptopts -p /sbin/mdev ${KOPT_debug_init:+-d} \
+if nlplug-findfs $cryptopts -p /sbin/mdev ${KOPT_debug_init:+-d} \
 	${KOPT_usbdelay:+-t $(( $KOPT_usbdelay * 1000 ))} \
-	$repoopts -a /tmp/apkovls
+	$repoopts -a /tmp/apkovls; then true
+elif [ -n "$KOPT_isoloop" ]; then for disk in /dev/sd*[0-9]; do
+	echo -n "Trying $disk... "
+	if mount -o ro "$disk" /media/usb 2> /dev/null; then
+		if mount -o loop,ro /media/usb/"$KOPT_isoloop" /media/cdrom 2> /dev/null; then
+			[ -n "$ALPINE_REPO" ] || echo /media/cdrom > "$repofile"
+			echo "succeeded"; break
+		else
+			umount /media/usb
+			echo "failed to mount iso"
+		fi
+	else echo "failed to mount disk"; fi
+	false
+done; else false; fi
```

Comment:
I don't think we should look for the isoloop this way. For example, what if the iso image is on a `/dev/mmc*`, `/dev/nvme*` or `/dev/cciss/*` device? or some other block device we currently don't know the name of? What if the isoloop is on a raid, lvm or crypt device?

We already solve that in `nlplug-findfs` so i think we should look for isoloop form there.

Reply:
You mean using UUID? Good idea, I will attempt to implement it later...